elastic/kibana
# Preserve originId when overwriting existing objects

Related to #135346 (partially resolves this issue).

### Overview

If an existing object is being overwritten (via create/bulkCreate), Kibana now preserves its originId field if it has one. This behavior can be overridden by specifying `originId: undefined`.

I also added validation to ensure that originId is only set for multi-namespace object types (it would basically have no effect for single-namespace object types, but we shouldn't allow it).

### Testing

1. Start Kibana
2. Add sample data
3. Create "Another" space
4. Copy a dashboard to Another space using the "Check for existing objects" option
5. In Another space, make a change to the dashboard and save it
6. In the Default space, copy the dashboard again using the "Check for existing objects" option
7. Observe that there are is still only one copy of the dashboard in Another space, and you can examine it to verify it still has the `originId` field

### Additional info

This does _not_ attempt to "repair" any saved objects that have had their `originId` field cleared out, it can be done in a separate PR, see the linked issue for more details.
